### PR TITLE
dev/user-interface#60 Add backward-compat to crmAccordionToggle function

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Makes older function work on new accordion markup, remove some unnecessarily complex handling for accordions & datatables.

Technical Details
---------
While providing backward-compat javascript support for our accordion => disclosure element upgrades, this also reverts 
16f4d319ef04e192a8836b2486f7de9dd7808046 to remove the unnecessary complexity of the 'crmAccordion:open' jQuery event.

### CODE REVIEW

**Comment by colemanw** [comment]
> @herbdool The `return` means it skips the last line: `$(this).toggleClass('collapsed').trigger('crmAccordion:' + action);`.

I think we should just get rid of that custom `crmAccordion:open` event. It's only used in 2 places. One of them was broken. [This fixes the broken one](https://github.com/civicrm/civicrm-core/pull/29452) (and swaps the `crmAccordion:open` event for a standard `click` which works fine).

Only one place left...

**Comment by colemanw** [comment]
Ok! https://github.com/civicrm/civicrm-core/pull/29452 has been merged and I've updated this PR to completely rip out the 'crmAccordion:open' jQuery event which I tracked down to a single commit that I added years ago as a nice-to-have-but-not-essential-thing.

If we're deprecating and ripping out accordions we definitely don't need to hang onto unnecessary bells & whistles.

File: js/Common.js [hunk 1]
```diff
@@ -1730,19 +1718,24 @@ if (!CRM.vars) CRM.vars = {};
...
+      if ($(this).is('details')) {
+        this.open = !this.open;
+        return;
+      }
       if ($(this).hasClass('collapsed')) {
         $('.crm-accordion-body', this).first().css('display', 'none').slideDown(speed);
       }
       else {
         $('.crm-accordion-body', this).first().css('display', 'block').slideUp(speed);
         action = 'close';
```
Comment (herbdool) [inline js/Common.js:1736]: Also delete this line?

Comment (colemanw) [reply js/Common.js:1736]: Yes, thanks

**Comment by herbdool** [comment]
I've tested this and it works.

I tested by converting the accordions on /contact/add to details and then changing the inline js to be able to find those details. Then this PR works as expected. The other changes are being addressed in https://github.com/civicrm/civicrm-core/pull/29448

**Pull request merged**